cmd/indexer/root.go

```diff
@@ -73,16 +79,7 @@ var ServerCmd = &cobra.Command{
 	Short: "blockchain data indexer",
 	Long:  `blockchain data indexer`,
 	RunE: func(cmd *cobra.Command, args []string) error {
-		vp := viper.New()
-		vp.BindPFlags(cmd.Flags())
-		vp.AutomaticEnv() // read in environment variables that match
-		if configFile := vp.GetString(flags.ConfigFileFlag); configFile != "" {
-			if err := loadConfigUsingViper(vp, configFile); err != nil {
-				log.Error("Failed to load config file", "err", err)
-				return err
-			}
-			loadFlagToVar(vp)
-		}
+		loadFlagToVar()
```

Comment:
`loadFlagToVar()` should be invoked in `initConfig()` because it load flags from config file. The flag values might be reset as initial value if we don't provide config file.

Reply:
Thanks! Updated in b98fbe353cddbf3c23e6cc1180e444bcbe9e42fd!